CIP-0025/README.md

```diff
@@ -64,27 +64,90 @@ The structure allows for multiple token mints, also with different policies, in
         <other properties>
       }
     },
-    "version": "1.0"
+    "version": <version_id>
   }
 }
 ```
 
-The **`asset_name`** must be `UTF-8` encoded for the key in the metadata map and the actual NFT. This is true for version `1.0`, future versions will use the `hex` encoding for **`asset_name`**.
+### CDDL
+
+#### Version 1
+
+```cddl
+
+policy_id = text
+asset_name = text ; utf-8
+
+files_details = 
+  {
+    name: text,
+    mediaType: text,
+    src: text / [* text]
+  }
+
+metadata_details = 
+  {
+    name : text,
+    image : text, 
+    ? mediaType: text,
+    ? description: text / [* text]
+    ? files: [* files_details]
+  }
+
+label_metadata = { * policy_id => { * asset_name => metadata_details } }
+
+metadata = { 721 : uint => label_metadata }
+```
+
+#### Version 2
+
+```cddl
+
+policy_id = bytes ; no longer in text
+asset_name = bytes ; no longer in text and utf-8
+
+files_details = 
+  {
+    name : text,
+    mediaType : text,
+    src : text / [* text]
+  }
+
+metadata_details = 
+  {
+    name : text,
```

Comment:
We have to be a little careful with this definition since text here isn't the standard "text" in cbor. Instead, Cardano metadata enforces that all text or byte array has a maximum length. We should probably either define these as a special type that matches the same max length or explicitly allow people to use an array to combine chunks together

Reply:
Yeah, I just fixed it. `name` should definitely stay below 64 characters. The ones that can exceed 64 characters have the type `string / [* string]`.